src/Validator/LdapUsernameValidator.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the package t3g/symfony-ldap-bundle.
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace T3G\Bundle\LdapBundle\Validator;
+
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use T3G\Bundle\LdapBundle\Service\LdapService;
+
+class LdapUsernameValidator extends ConstraintValidator
+{
+    /**
+     * @var LdapService
+     */
+    private $ldapService;
+
+    public function __construct(LdapService $ldapService)
+    {
+        $this->ldapService = $ldapService;
+    }
+
+    public function validate($value, Constraint $constraint)
+    {
+        /* @var $constraint LdapUsername */
```

Comment:
I would use a guard clause here like in all other symfony validators, checking the Constraint type:

```
if (!$constraint instanceof LdapUsername) {
    throw new UnexpectedTypeException($constraint, LdapUsername::class);
}
```